Software/Development/SemesterProject/App/temperatureManager.c

```diff
@@ -0,0 +1,49 @@
+/*
+ * temperatureManager.c
+ *
+ *  Created on: Apr 8, 2019
+ *      Author: Nicolas
+ *
+ *      The temperature manager is a module that obtains the voltage measurement
+ *      from the analogAcquisitionManager  and calculates the temperature of the sensor.
+ *      It also provides an interface for external access to the temperature.
+ *
+ *      calculateTemperature is called every second from the scheduler.
+ *
+ *      The temperature sensors were not unified in a higher level of abstraction
+ *      structure because the number of sensors is two and it not likely to change.
+ */
+
+#include "temperatureManager.h"
+
+/*
+ * Quasi Global variable definition
+ */
+static float S1Temperature = 0, S2Temperature = 0;
+
+void calculateTemperature(void) //To be called from scheduler when reference handler is merged
+{
+    float voltageSensor1;
+    voltageSensor1 = getThermometer1Measurement();
+    S1Temperature = voltageSensor1 / TEMP_SENSOR_GAIN;
```

Comment:
If instead of creating the gain as mV/deg you do deg/mV in constants, then you can have a multiplication by an int instead of a division of floats. No big deal

Reply:
Nice idea!